axon-server-connector/src/main/java/org/axonframework/axonserver/connector/processor/EventProcessorControlService.java

```diff
@@ -146,9 +158,13 @@ private void splitSegment(PlatformOutboundInstruction platformOutboundInstructio
         int segmentId = splitSegment.getSegmentIdentifier();
         String processorName = splitSegment.getProcessorName();
         try {
-            eventProcessorController.splitSegment(processorName, segmentId);
+            if (!eventProcessorController.splitSegment(processorName, segmentId)) {
```

Comment:
The `EventProcessorController#splitSegment(String, int)` function will also return `false` if the underlying `TrackingEventProcessor` couldn't fulfill the request. Thus stating it's an `IllegalArgumentException` because the given `processorName` is not a `TrackingEventProcessor` is not always correct.

Reply:
Yes, you are right. I will change it with a generic message like "Impossible to split/merge segment [{}] for processor [{}]", or something similar.